### DISCUSSION TITLE
Transition to UTouch (Universal Touch) driver

### DISCUSSION THREAD

**Discussion by Jason2866** [arendst/Tasmota#21039] [Ideas]
Follow up of actually open todo's (copy from PR) identified by @gemu2015

- the CSx touch drivers are not yet tested but should be easy, the gt911 driver has a special issue.
these ICs have to be initially flashed with a config block for several parameters like horizontal and vertical resolution.
normally this should only happen once in the displays lifetime. the legacy drivers compares the resolution of the display and the touch ic on startup and on mismatch flashes the chip. i did not implement this in universal driver, so may be we should have a berry script to flash that chip if it ever should be necessary
- some families of drivers (like FTx) have the same register assignments and so the code is identical, but the chip ID is different.
currently we compare the chip id on startup. i will provide a multi compare opcode to match several ICs later
- i would not remove other touch drivers now, but wait until more people have tested u-touch and all display.ini files have been updated. simply do not define them and make u-touch the default.
-> done, UTouch is enabled by default in build lvgl

**Comment by s-hadinger** [comment]
Thanks @Jason2866. I'm interested to get confirmation than enabling uTouch does not break legacy drivers.

**Comment by s-hadinger** [comment]
I have isolated all information related to migrating to uDisplay and uTouch here: https://tasmota.github.io/docs/Universal-Display-Driver/

@gemu2015 What do you think about using this page for uDisplay documentation? The current one is hard to find within the Displays page

**Reply by gemu2015** [reply]
Yes i agree! 👍

**Reply by s-hadinger** [reply]
It is completely moved now
